internal/agent/types/types.go

```diff
@@ -0,0 +1,36 @@
+package types
```

Comment:
Yes, you would face import cycle and would required an even bigger refactor than this change. I suggest keeping like this for this PR and when we cleanup old code we would be able to avoid this issue

Reply:
👍